### PR TITLE
update sphinx gallery to 0.1.8

### LINKED ISSUE #2410: Update sphinx-gallery before release 0.13
As we patched sphinx-gallery embeded in scikit-image, we need to update the plugin before the release, either against their master branch or with their new release (should come soon).

See https://github.com/scikit-image/scikit-image/pull/2395

* A line has been added in TODO.txt. The task has to be removed from that file.
* Waiting for merge of https://github.com/sphinx-gallery/sphinx-gallery/pull/190
* Waiting for 0.1.8 or larger: https://github.com/sphinx-gallery/sphinx-gallery/releases

### LINKED ISSUE #2481: Empty orphan boxes in our gallery
Reported upstream: https://github.com/sphinx-gallery/sphinx-gallery/issues/202

### PR DESCRIPTION
**Pull request by sciunto**
Closes #2410
Superseed #2457 

Should fix #2481 

I don't think we need two reviewers, it just a merge.

### CODE REVIEW

**Comment by sciunto** [comment]
The failure with py2.7 and the min req is not due to this PR. See #2476

File: doc/ext/sphinx_gallery/downloads.py [hunk 1]
```diff
@@ -55,14 +55,14 @@ def python_zip(file_list, gallery_path, extension='.py'):
         zip file name, written as `target_dir_{python,jupyter}.zip`
         depending on the extension
     """
-    zipname = gallery_path.replace(os.path.sep, '_')
+    zipname = os.path.basename(gallery_path)
```
Comment (jni) [inline doc/ext/sphinx_gallery/downloads.py]: This seems like a significant change... What's the effect of this?

**Comment by sciunto** [comment]
@jni It's just a merge in the sense that I copy pasted their code. We use it as a plugin. We are not supposed to review the code, their review the code by themselves. In the future, we should handle this plugin better

**Comment by jni** [comment]
Ah! fair enough. Have you run it locally to make sure it didn't break anything?

**Comment by sciunto** [comment]
Checked. And it will also close #2481 as far as I can see on my laptop.

**Comment by jni** [comment]
Done, then! =)

**Pull request merged**